src/overlays/actors/ovl_Fishing/z_fishing.c

```diff
@@ -216,97 +325,5459 @@ static ColliderJntSphInit D_80B7A888 = {
+            sReelLinePos[LINE_SEG_COUNT - 1].z = sLurePos.z;
+            D_80B7E148 = 2.0f;
+            break;
+    }
+}
+
+s32 func_80B70A2C(Fishing* this, GlobalContext* globalCtx, u8 ignorePosCheck) {
+    s16 i;
+    s16 count;
+    f32 scale;
+    Vec3f pos;
+    Vec3f vel;
+    f32 speedXZ;
+    f32 angle;
+
+    if ((this->actor.world.pos.y < (WATER_SURFACE_Y(globalCtx) - 10.0f)) && !ignorePosCheck) {
+        return false;
+    }
+
+    // Necessary to match
+    if (this->unk_1AC) {}
+
+    if (this->unk_1AC >= 40.0f) {
+        count = 40;
+        scale = 1.2f;
+    } else {
+        count = 30;
+        scale = 1.0f;
+    }
+
+    for (i = 0; i < count; i++) {
+        speedXZ = (Rand_ZeroFloat(1.5f) + 0.5f) * scale;
+        angle = Rand_ZeroFloat(6.28f);
+
+        vel.x = sinf(angle) * speedXZ;
+        vel.z = cosf(angle) * speedXZ;
+        vel.y = (Rand_ZeroFloat(3.0f) + 3.0f) * scale;
+
+        pos = this->actor.world.pos;
+        pos.x += vel.x * 3.0f;
+        pos.y = WATER_SURFACE_Y(globalCtx);
+        pos.z += vel.z * 3.0f;
+
+        Fishing_SpawnDustSplash(&this->actor.projectedPos, globalCtx->specialEffects, &pos, &vel,
+                                (Rand_ZeroFloat(0.02f) + 0.025f) * scale);
+    }
+
+    pos = this->actor.world.pos;
+    pos.y = WATER_SURFACE_Y(globalCtx);
+
+    Fishing_SpawnRipple(&this->actor.projectedPos, globalCtx->specialEffects, &pos, 100.0f, 800.0f, 150, 90);
+
+    this->unk_151 = 30;
+
+    return true;
+}
+
+void func_80B70CF0(Fishing* this, GlobalContext* globalCtx) {
+    s16 count;
+    s16 i;
+    f32 scale;
+    Vec3f pos;
+    Vec3f vel;
+    f32 speedXZ;
+    f32 angle;
+
+    // Necessary to match
+    if (this->unk_1AC) {}
+
+    if (this->unk_1AC >= 45.0f) {
+        count = 30;
+        scale = 0.5f;
+    } else {
+        count = 20;
+        scale = 0.3f;
+    }
+
+    for (i = 0; i < count; i++) {
+        speedXZ = (Rand_ZeroFloat(1.5f) + 0.5f) * scale;
+        angle = Rand_ZeroFloat(6.28f);
+
+        vel.x = sinf(angle) * speedXZ;
+        vel.z = cosf(angle) * speedXZ;
+        vel.y = Rand_ZeroFloat(2.0f) + 2.0f;
+
+        pos = this->actor.world.pos;
+        pos.x += (vel.x * 3.0f);
+        pos.y += (vel.y * 3.0f);
+        pos.z += (vel.z * 3.0f);
+
+        Fishing_SpawnDustSplash(&this->actor.projectedPos, globalCtx->specialEffects, &pos, &vel,
+                                (Rand_ZeroFloat(0.02f) + 0.025f) * scale);
+    }
+}
+
+void func_80B70ED4(Fishing* this, Input* input) {
+    Vec3f sp34;
+    Vec3f sp28;
+    f32 sp24;
+
+    sp34.x = sLurePos.x - this->actor.world.pos.x;
+    sp34.y = sLurePos.y - this->actor.world.pos.y;
+    sp34.z = sLurePos.z - this->actor.world.pos.z;
+
+    sp24 = SQ(sp34.x) + SQ(sp34.y) + SQ(sp34.z);
+
+    if ((D_80B7A694 == 3) && (this->unk_1A2 == 0) && (D_80B7A68C == 0)) {
+        Matrix_RotateY(((0 - this->actor.shape.rot.y) / 32768.0f) * M_PI, MTXMODE_NEW);
```

Comment:
```suggestion
        Matrix_RotateY(-this->actor.shape.rot.y / (f32)0x8000 * M_PI, MTXMODE_NEW);
```
and others `32768.0f` in the file, some have to be casted some don't